src/__tests__/helpers.js

```diff
@@ -1,9 +1,53 @@
-import {getDocument, newMutationObserver} from '../helpers'
+import {
+  getDocument,
+  newMutationObserver,
+  getSetTimeout,
+  getClearTimeout,
+} from '../helpers'
 
 test('returns global document if exists', () => {
   expect(getDocument()).toBe(document)
 })
 
+describe('getSetTimeout', () => {
+  it('returns mocked setTimeout if global.useFakeTimers is set and jest.useFakeTimers is used', () => {
+    // global.useFakeTimers is set to true for all tests in tests/setup-env.js
```

Comment:
If that's the case, then isn't `jest.useFakeTimers()` on the next line unnecessary and `jest.useRealTimers()` at the end of this test breaking test isolation?

Reply:
Actually, now that I think of it, why are we setting `useFakeTimers` to true in setup-env? I think it'd be better if most of our tests run without fake timers wouldn't it?

Reply:
global.useFakeTimers is a boolean that enables fake timers **if** someone uses jest.useFakeTimers(). So most of our tests will be run without fake timers.

I think we can improve upon the name of the boolean. How about global.returnMockedTimers? Or would you prefer that we leave this undefined by default and only set this to true in our tests that use fake timers (https://github.com/testing-library/dom-testing-library/search?q=jest.useFakeTimers&unscoped_q=jest.useFakeTimers)?

Reply:
Oh wait. Are you saying that if people want to take advantage of this feature they have to set `global.useFakeTimers`? I'd really prefer to find another way to detect whether fake timers are happening... Maybe I don't understand what's going on/what the purpose of this is 😬

Reply:
Nop, nop, exactly the opposite! This behaviour is enabled by default for everyone who uses dom-testing-library in their application tests, but it's disabled for dom-testing-library internal tests.

I do think that I'm explaining and documenting this rather poorly. Let me explain it to you first with a bunch of examples, I hope that clears things up! Perhaps than you give me some pointers on how exactly I can communicate this behaviour better? Your suggestions are very much welcome!

If I as a user use dom-testing-library all the dom-testing library methods that I use in my application tests will use a non-mocked setTimeout. So if I write:

```
jest.useFakeTimers();
findByText('My application is awesome!')
```

dom-testing-library will check for 4.5 seconds if the text 'Awesome Text' is in my application.

Without this PR, the above test will most likely fail because findByText uses a mocked fake timer which reduces the timeout to 1 tick.

If I as a contributor for dom-testing-library write a new test for dom-testing-library and if I write the following test:

````
jest.useFakeTimers();
setTimeOut(() => {
   findByText('Dom testing library is awesome!')
, 9999}
````

This test will use the mocked setTimeout and thus succeed within <10 msecs. This is exactly the same behaviour as before this PR.

The global is used to determine if we're running the dom-testing-library internal tests.

I hope this clears things up :)

The [fix ](https://github.com/TheBrainFamily/wait-for-expect/pull/5/files) for wait-for-expect is a lot smaller because they don't have any internal tests that use jest.useFakeTimers() which greatly reduces the complexity.

Reply:
Oh, I see what you're saying. So the whole reason that we have the global thing is for internal testing purposes.

Hmm... I'm not sure I understand why it's necessary though. Why can't DOM Testing Library's tests work the same way that users of DTL test?

Reply:
Ok, I think I've worked it out. I'm going to make a contribution to this to improve things I think.